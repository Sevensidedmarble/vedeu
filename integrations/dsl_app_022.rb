#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = 'dsl_app_022'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    height 10
    log Dir.tmpdir + '/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename: Dir.tmpdir + "/#{TESTCASE}.out",
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: Dir.tmpdir + "/#{TESTCASE}.json"),
                # Vedeu::Renderers::HTML.new(filename: Dir.tmpdir + "/#{TESTCASE}.html"),
                # Vedeu::Renderers::Text.new(filename: Dir.tmpdir + "/#{TESTCASE}.txt"),
              ]
    run_once!
    standalone!
    height 50
    width 100
  end

  load File.dirname(__FILE__) + '/support/test_interface_022.rb'

  Vedeu.render do
    view(:test_interface) do
      line do
        stream '2--'
        stream '+', foreground: '#ffff00'
        stream '----10--------20--------30--------40--------50'
        stream '--------60--------70--------80---'
        stream '+', foreground: '#ffff00'
        stream '----90---'
        stream '+', foreground: '#ffff00'
        stream '----100'
        stream '-------10--------20--------30--------40-------50'
      end
      line do
        stream '- '
        stream 'geometry height: 40', foreground: '#aadd00'
      end
      line do
        stream '- '
        stream 'geometry width: 90', foreground: '#aadd00'
      end
      line '+', foreground: '#ffff00'
      line '-'
      line '-'
      line '-'
      line '-'
      line '10'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '20'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '30'
      line '-'
      line '-'
      line '-'
      line '-'
      line '+', foreground: '#ffff00'
      line '-'
      line '-'
      line '-'
      line '-'
      line '40'
      line '-'
      line '-'
      line '-'
      line '-'
      line '+', foreground: '#ffff00'
      line '-'
      line '-'
      line '-'
      line '-'
      line '50'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '-'
      line '60'
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

Vedeu.timer('Test') do
  DSLApp.start
end

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
