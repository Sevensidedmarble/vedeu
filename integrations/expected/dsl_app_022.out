[0m[1;1H[38;5;231m[48;5;18m(0l(B[1;2H(0q(B[1;3H [1;4HW[1;5Hi[1;6Hd[1;7Ht[1;8Hh[1;9H:[1;10H [1;11H1[1;12H0[1;13H0[1;14H [1;15HH[1;16He[1;17Hi[1;18Hg[1;19Hh[1;20Ht[1;21H:[1;22H [1;23H5[1;24H0[1;25H [1;26H(0q(B[1;27H(0q(B[1;28H(0q(B[1;29H(0q(B[1;30H(0q(B[1;31H(0q(B[1;32H(0q(B[1;33H(0q(B[1;34H(0q(B[1;35H(0q(B[1;36H(0q(B[1;37H(0q(B[1;38H(0q(B[1;39H(0q(B[1;40H(0q(B[1;41H(0q(B[1;42H(0q(B[1;43H(0q(B[1;44H(0q(B[1;45H(0q(B[1;46H(0q(B[1;47H(0q(B[1;48H(0q(B[1;49H(0q(B[1;50H(0q(B[1;51H(0q(B[1;52H(0q(B[1;53H(0q(B[1;54H(0q(B[1;55H(0q(B[1;56H(0q(B[1;57H(0q(B[1;58H(0q(B[1;59H(0q(B[1;60H(0q(B[1;61H(0q(B[1;62H(0q(B[1;63H(0q(B[1;64H(0q(B[1;65H(0q(B[1;66H(0q(B[1;67H(0q(B[1;68H(0q(B[1;69H(0q(B[1;70H(0q(B[1;71H(0q(B[1;72H(0q(B[1;73H(0q(B[1;74H(0q(B[1;75H(0q(B[1;76H(0q(B[1;77H(0q(B[1;78H(0q(B[1;79H(0q(B[1;80H(0q(B[1;81H(0q(B[1;82H(0q(B[1;83H(0q(B[1;84H(0q(B[1;85H(0q(B[1;86H(0q(B[1;87H(0q(B[1;88H(0q(B[1;89H(0q(B[1;90H(0k(B[0m[2;1H[38;5;231m[48;5;18m(0x(B[2;2H2[2;3H-[2;4H-[2;5H[38;5;226m[48;5;18m+[2;6H[38;5;231m[48;5;18m-[2;7H-[2;8H-[2;9H-[2;10H1[2;11H0[2;12H-[2;13H-[2;14H-[2;15H-[2;16H-[2;17H-[2;18H-[2;19H-[2;20H2[2;21H0[2;22H-[2;23H-[2;24H-[2;25H-[2;26H-[2;27H-[2;28H-[2;29H-[2;30H3[2;31H0[2;32H-[2;33H-[2;34H-[2;35H-[2;36H-[2;37H-[2;38H-[2;39H-[2;40H4[2;41H0[2;42H-[2;43H-[2;44H-[2;45H-[2;46H-[2;47H-[2;48H-[2;49H-[2;50H5[2;51H0[2;52H-[2;53H-[2;54H-[2;55H-[2;56H-[2;57H-[2;58H-[2;59H-[2;60H6[2;61H0[2;62H-[2;63H-[2;64H-[2;65H-[2;66H-[2;67H-[2;68H-[2;69H-[2;70H7[2;71H0[2;72H-[2;73H-[2;74H-[2;75H-[2;76H-[2;77H-[2;78H-[2;79H-[2;80H8[2;81H0[2;82H-[2;83H-[2;84H-[2;85H[38;5;226m[48;5;18m+[2;86H[38;5;231m[48;5;18m-[2;87H-[2;88H-[2;89H-[2;90H(0x(B[0m[3;1H[38;5;231m[48;5;18m(0x(B[3;2H-[3;3H [3;4H[38;5;148m[48;5;18mg[3;5He[3;6Ho[3;7Hm[3;8He[3;9Ht[3;10Hr[3;11Hy[3;12H [3;13Hh[3;14He[3;15Hi[3;16Hg[3;17Hh[3;18Ht[3;19H:[3;20H [3;21H4[3;22H0[3;90H[38;5;231m[48;5;18m(0x(B[0m[4;1H[38;5;231m[48;5;18m(0x(B[4;2H-[4;3H [4;4H[38;5;148m[48;5;18mg[4;5He[4;6Ho[4;7Hm[4;8He[4;9Ht[4;10Hr[4;11Hy[4;12H [4;13Hw[4;14Hi[4;15Hd[4;16Ht[4;17Hh[4;18H:[4;19H [4;20H9[4;21H0[4;90H[38;5;231m[48;5;18m(0x(B[0m[5;1H[38;5;231m[48;5;18m(0x(B[5;2H[38;5;226m[48;5;18m+[5;90H[38;5;231m[48;5;18m(0x(B[0m[6;1H[38;5;231m[48;5;18m(0x(B[6;2H-[6;90H(0x(B[0m[7;1H[38;5;231m[48;5;18m(0x(B[7;2H-[7;90H(0x(B[0m[8;1H[38;5;231m[48;5;18m(0x(B[8;2H-[8;90H(0x(B[0m[9;1H[38;5;231m[48;5;18m(0x(B[9;2H-[9;90H(0x(B[0m[10;1H[38;5;231m[48;5;18m(0x(B[10;2H1[10;3H0[10;90H(0x(B[0m[11;1H[38;5;231m[48;5;18m(0x(B[11;2H-[11;90H(0x(B[0m[12;1H[38;5;231m[48;5;18m(0x(B[12;2H-[12;90H(0x(B[0m[13;1H[38;5;231m[48;5;18m(0x(B[13;2H-[13;90H(0x(B[0m[14;1H[38;5;231m[48;5;18m(0x(B[14;2H-[14;90H(0x(B[0m[15;1H[38;5;231m[48;5;18m(0x(B[15;2H-[15;90H(0x(B[0m[16;1H[38;5;231m[48;5;18m(0x(B[16;2H-[16;90H(0x(B[0m[17;1H[38;5;231m[48;5;18m(0x(B[17;2H-[17;90H(0x(B[0m[18;1H[38;5;231m[48;5;18m(0x(B[18;2H-[18;90H(0x(B[0m[19;1H[38;5;231m[48;5;18m(0x(B[19;2H-[19;90H(0x(B[0m[20;1H[38;5;231m[48;5;18m(0x(B[20;2H2[20;3H0[20;90H(0x(B[0m[21;1H[38;5;231m[48;5;18m(0x(B[21;2H-[21;90H(0x(B[0m[22;1H[38;5;231m[48;5;18m(0x(B[22;2H-[22;90H(0x(B[0m[23;1H[38;5;231m[48;5;18m(0x(B[23;2H-[23;90H(0x(B[0m[24;1H[38;5;231m[48;5;18m(0x(B[24;2H-[24;90H(0x(B[0m[25;1H[38;5;231m[48;5;18m(0x(B[25;2H-[25;90H(0x(B[0m[26;1H[38;5;231m[48;5;18m(0x(B[26;2H-[26;90H(0x(B[0m[27;1H[38;5;231m[48;5;18m(0x(B[27;2H-[27;90H(0x(B[0m[28;1H[38;5;231m[48;5;18m(0x(B[28;2H-[28;90H(0x(B[0m[29;1H[38;5;231m[48;5;18m(0x(B[29;2H-[29;90H(0x(B[0m[30;1H[38;5;231m[48;5;18m(0x(B[30;2H3[30;3H0[30;90H(0x(B[0m[31;1H[38;5;231m[48;5;18m(0x(B[31;2H-[31;90H(0x(B[0m[32;1H[38;5;231m[48;5;18m(0x(B[32;2H-[32;90H(0x(B[0m[33;1H[38;5;231m[48;5;18m(0x(B[33;2H-[33;90H(0x(B[0m[34;1H[38;5;231m[48;5;18m(0x(B[34;2H-[34;90H(0x(B[0m[35;1H[38;5;231m[48;5;18m(0x(B[35;2H[38;5;226m[48;5;18m+[35;90H[38;5;231m[48;5;18m(0x(B[0m[36;1H[38;5;231m[48;5;18m(0x(B[36;2H-[36;90H(0x(B[0m[37;1H[38;5;231m[48;5;18m(0x(B[37;2H-[37;90H(0x(B[0m[38;1H[38;5;231m[48;5;18m(0x(B[38;2H-[38;90H(0x(B[0m[39;1H[38;5;231m[48;5;18m(0x(B[39;2H-[39;90H(0x(B[0m[40;1H[38;5;231m[48;5;18m(0m(B[40;2H(0q(B[40;3H(0q(B[40;4H(0q(B[40;5H(0q(B[40;6H(0q(B[40;7H(0q(B[40;8H(0q(B[40;9H(0q(B[40;10H(0q(B[40;11H(0q(B[40;12H(0q(B[40;13H(0q(B[40;14H(0q(B[40;15H(0q(B[40;16H(0q(B[40;17H(0q(B[40;18H(0q(B[40;19H(0q(B[40;20H(0q(B[40;21H(0q(B[40;22H(0q(B[40;23H(0q(B[40;24H(0q(B[40;25H(0q(B[40;26H(0q(B[40;27H(0q(B[40;28H(0q(B[40;29H(0q(B[40;30H(0q(B[40;31H(0q(B[40;32H(0q(B[40;33H(0q(B[40;34H(0q(B[40;35H(0q(B[40;36H(0q(B[40;37H(0q(B[40;38H(0q(B[40;39H(0q(B[40;40H(0q(B[40;41H(0q(B[40;42H(0q(B[40;43H(0q(B[40;44H(0q(B[40;45H(0q(B[40;46H(0q(B[40;47H(0q(B[40;48H(0q(B[40;49H(0q(B[40;50H(0q(B[40;51H(0q(B[40;52H(0q(B[40;53H(0q(B[40;54H(0q(B[40;55H(0q(B[40;56H(0q(B[40;57H(0q(B[40;58H(0q(B[40;59H(0q(B[40;60H(0q(B[40;61H(0q(B[40;62H(0q(B[40;63H(0q(B[40;64H(0q(B[40;65H(0q(B[40;66H(0q(B[40;67H(0q(B[40;68H(0q(B[40;69H(0q(B[40;70H(0q(B[40;71H(0q(B[40;72H(0q(B[40;73H(0q(B[40;74H(0q(B[40;75H(0q(B[40;76H(0q(B[40;77H(0q(B[40;78H(0q(B[40;79H(0q(B[40;80H(0q(B[40;81H(0q(B[40;82H(0q(B[40;83H(0q(B[40;84H(0q(B[40;85H(0q(B[40;86H(0q(B[40;87H(0q(B[40;88H(0q(B[40;89H(0q(B[40;90H(0j(B[0m